Source/WebCore/page/csp/ContentSecurityPolicy.cpp

```diff
@@ -806,9 +806,8 @@ bool ContentSecurityPolicy::allowMissingTrustedTypesForSinkGroup(const String& s
             String consoleMessage = makeString(policy->isReportOnly() ? "[Report Only] "_s : ""_s,
                 "This requires a "_s, stringContext, " value else it violates the following Content Security Policy directive: \"require-trusted-types-for 'script'\""_s);
 
-            TextPosition sourcePosition(OrdinalNumber::beforeFirst(), OrdinalNumber());
-            String sample = makeString(sink, '|', source);
-            reportViolation("require-trusted-types-for"_s, *policy, "trusted-types-sink"_s, consoleMessage, nullString(), sample, sourcePosition, nullptr);
+            String sample = makeString(sink, '|', source.left(40));
```

Comment:
Since source is a `String`, for efficiency this should say `StringView(source).left(40)` to avoid wastefully allocating and destroying a temporary `StringImpl`. Better, the argument type should be changed to `StringView` instead. Many of these arguments should be changed to `StringView`. We’d only take `const String&` if we are storing that string as-is or passing it along to another function that requires a `String` and never take `const StringView&`.

Reply:
I've changed the method to take  StringView instead.